aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu

```diff
@@ -377,6 +377,1043 @@ void entr_kernel_cuda(TensorIteratorBase& iter) {
   #endif
 }
 
+const auto prime_number_string = jiterator_stringify(
+template<typename T1>
+C10_HOST_DEVICE
+inline constexpr
+T1
+prime_number(unsigned int n) {
+  constexpr std::uint16_t prime_numbers[10000] = {
+      0x000002U, 0x000003U, 0x000005U, 0x000007U, 0x00000bU, 0x00000dU, 0x000011U, 0x000013U, 0x000017U, 0x00001dU,
+      0x00001fU, 0x000025U, 0x000029U, 0x00002bU, 0x00002fU, 0x000035U, 0x00003bU, 0x00003dU, 0x000043U, 0x000047U,
+      0x000049U, 0x00004fU, 0x000053U, 0x000059U, 0x000061U, 0x000065U, 0x000067U, 0x00006bU, 0x00006dU, 0x000071U,
+      0x00007fU, 0x000083U, 0x000089U, 0x00008bU, 0x000095U, 0x000097U, 0x00009dU, 0x0000a3U, 0x0000a7U, 0x0000adU,
+      0x0000b3U, 0x0000b5U, 0x0000bfU, 0x0000c1U, 0x0000c5U, 0x0000c7U, 0x0000d3U, 0x0000dfU, 0x0000e3U, 0x0000e5U,
+      0x0000e9U, 0x0000efU, 0x0000f1U, 0x0000fbU, 0x000101U, 0x000107U, 0x00010dU, 0x00010fU, 0x000115U, 0x000119U,
+      0x00011bU, 0x000125U, 0x000133U, 0x000137U, 0x000139U, 0x00013dU, 0x00014bU, 0x000151U, 0x00015bU, 0x00015dU,
+      0x000161U, 0x000167U, 0x00016fU, 0x000175U, 0x00017bU, 0x00017fU, 0x000185U, 0x00018dU, 0x000191U, 0x000199U,
+      0x0001a3U, 0x0001a5U, 0x0001afU, 0x0001b1U, 0x0001b7U, 0x0001bbU, 0x0001c1U, 0x0001c9U, 0x0001cdU, 0x0001cfU,
+      0x0001d3U, 0x0001dfU, 0x0001e7U, 0x0001ebU, 0x0001f3U, 0x0001f7U, 0x0001fdU, 0x000209U, 0x00020bU, 0x00021dU,
```

Comment:
Please don't do that, can't you rather generate this string from the table you've already had on cpu?

Reply:
How would you accomplish this (i.e., transform the declaration to a string)? Is there an example I could look at?